task_2/task_2.txt

```diff
@@ -0,0 +1,23 @@
+There are several possible ways to provide the secrets to running pods securely:
+
+1. AWS KMS envelope encryption https://aws.amazon.com/blogs/containers/using-eks-encryption-provider-support-for-defense-in-depth/;
++ completely transparent to the application;
++ does not break common flow, secrets are referenced as usual;
++ no additional configuration is needed;
+
+2. kubernetes-external-secrets which creates ExternalSecrets crd, pulling them from AWS Parameter Store or SecretsManager https://github.com/godaddy/kubernetes-external-secrets 
++ data is encrypted and self documented;
+- onetime configuration is needed;
+
+3. access SSM Parameter Store directly from the application via AWS api call;
++ data is encrypted and self documented;
+- might become cumbersome to manage when number of apps is growing, each app will require own IAM role;
```

Comment:
Let's assume that there is a requirement to assign an IAM role to each subset of pods, unrelated to the secrets management solution - how would you implement that in EKS? how the roles are going to be created and how are they going to be assigned to the running pods?

Reply:
One possible solution is to use terraform to generate IAM roles and policies with terraform like in https://github.com/eugenetaranov/eks_cluster_tf/blob/task_1/task_1/cluster_tf/testservice.tf_#L28, also terraform can generate service accounts in kubernetes. Once this is ready, serviceAccountName (service account generated by terraform) reference should be included into pod declaration.